### PR TITLE
NW-6368 | Mirror kubexit & add Acquia pipeline

### PR DESCRIPTION
**Pull request by renishb10**
### Manual Test
- CI-CD build should trigger and successfully built
- Image found on the Jfrog artifactory

### CODE REVIEW

File: .acquia/pipeline.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+pre_build:
+  security_composition_analysis:
+    required: false
+    language:
+    - go
+  clone_source_image:
+  - steps:
+    - | 
+      export KUBEXIT_VERSION=v0.3.2-acquia7
+      git clone --depth 1 --branch "${KUBEXIT_VERSION}" https://github.com/karlkfi/kubexit.git mirror
```
Comment (acquiaanil) [inline .acquia/pipeline.yaml:28]: Why we are cloning/using `karlkfi/kubexit`? I believe `acquia/kubexit` has additions (see history).
I see rest of the files (source) has been deleted.

Comment (renishb10) [reply .acquia/pipeline.yaml:28]: Yes, I believe, we need to clone the actual repo and apply the history (only bug fix - tombstone one), as git patch.
Just like how we did for other repo, fn-apache-exporter, fn-memcached-exporter etc.

cc @alexxed

Comment (alexxed) [reply .acquia/pipeline.yaml:28]: we're changing this to not be a fork, but to work with our regular mirror process:
- clone from source
- apply any patches
- build image

Comment (acquiaanil) [reply .acquia/pipeline.yaml:28]: @renishb10  
Where are the step(s) for `apply the history (only bug fix - tombstone one), as git patch`? 
I'm not sure how we will add fixes or feature / update go.mod in `acquia/kubexit` in future.

**Comment by renishb10** [comment]
Image pushed to Jfrog
```
> docker pull jfrog.ais.acquia.io/devops-pipeline/fn-kubexit:latest
latest: Pulling from devops-pipeline/fn-kubexit
213ec9aee27d: Already exists
2312baab98fd: Pull complete
adaa7930342c: Pull complete
Digest: sha256:f037970a6c47d027e0cfca8c1cb19f5602fd626e9477f15841ad491d3da6457e
Status: Downloaded newer image for jfrog.ais.acquia.io/devops-pipeline/fn-kubexit:latest
jfrog.ais.acquia.io/devops-pipeline/fn-kubexit:latest
```

File: README.md [hunk 2]
```diff
@@ -1,113 +1,13 @@
...
-  - mountPath: /kubexit
-    name: kubexit
-```
-
-## Examples
-
-- [Client Server Job](examples/client-server-job/)
-- [CloudSQL Proxy Job](examples/cloudsql-proxy-job/)
+docker run --name fn-kubexit
+```
```
Comment (acquiaanil) [inline README.md:13]: If applicable, please add instructions to update the dependencies/logic.

File: .acquia/Dockerfile.ci [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+RUN curl -LO "https://dl.k8s.io/release/${K8S_VERSION}/bin/linux/amd64/kubectl"
+RUN install -o root -g root -m 0755 kubectl /usr/local/bin/kubectl
+
+# Install k3d
+RUN curl -s https://raw.githubusercontent.com/rancher/k3d/main/install.sh | TAG=${K3D_VERSION} bash
+
+# Install helm
+RUN curl https://get.helm.sh/helm-${HELM_VERSION}-linux-amd64.tar.gz | tar -xz
+RUN mv linux-amd64/helm /usr/local/bin
```
Comment (acquiaanil) [inline .acquia/Dockerfile.ci:50]: nit : new line

File: .acquia/pipeline.yaml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+      - steps:
+          - echo Required for jfrog artifact promotion.
+    staging:
+      - steps:
+          - echo Required for jfrog artifact promotion.
+
+notify:
+  channel: proj-ngc
+  on_success: change
+  on_failure: always
```
Comment (acquiaanil) [inline .acquia/pipeline.yaml:83]: nit : new line

**Review by acquiaanil — APPROVED** [review]
Looks good to me. 
Only doubt is that how do we update the dependencies / source because we are managing the forked.

File: .acquia/pipeline.yaml (hunk 1 above)
Comment (alexxed) [reply .acquia/pipeline.yaml:28]: in separate patches files, e.g. patch1, patch2 in the root dir of the repo

File: README.md (hunk 2 above)
Comment (alexxed) [reply README.md:13]: Yeah, add instructions to:
- add more patches
- update the base version

File: .acquia/pipeline.yaml (hunk 1 above)
Comment (renishb10) [reply .acquia/pipeline.yaml:28]: I've added a directory named `/patches` and added wildcard to the git apply cmd
```
git apply --reject --whitespace=fix ../patches/*.patch
```

File: README.md (hunk 2 above)
Comment (renishb10) [reply README.md]: Updated https://github.com/acquia/kubexit/blob/d5e2c57fc89fae93f34efeafa2096c0ee3c70789/README.md

**Comment by renishb10** [comment]
> Looks good to me. Only doubt is that how do we update the dependencies / source because we are managing the forked.

Thanks Anil. Updated Readme. https://github.com/acquia/kubexit/blob/d5e2c57fc89fae93f34efeafa2096c0ee3c70789/README.md

File: README.md [hunk 5]
```diff
@@ -1,113 +1,13 @@
...
-  volumeMounts:
-  - mountPath: /kubexit
-    name: kubexit
-```
-
-## Examples
-
-- [Client Server Job](examples/client-server-job/)
-- [CloudSQL Proxy Job](examples/cloudsql-proxy-job/)
+docker run --name fn-kubexit
```
[the remark this answers is not in the document]
Comment (renishb10) [reply README.md]: updated

**Review by alexxed — APPROVED** [review]

**Pull request merged**